docs/06-actors-actions.md

```diff
@@ -0,0 +1,414 @@
+
+The benefit of something like a state machine is that state can be separated from the actions an actor may take and you can then _save_ and _load_ state more easily to enable save game management. You could choose for example to have a global game state that you can serialize and deserialize.
+
+<docs-note>Have you implemented state management in your Excalibur game? [Let us know](https://github.com/excaliburjs/Excalibur#questions)!</docs-note>
+
+## Update hooks
+
+There are three ways to hook into the update loop of an actor: [[Actor.onPreUpdate]], [[Actor.update]] and [[Actor.onPostUpdate]]. Actors (and other entities in Excalibur) all have "core" logic that runs in the update or draw loop. The pre- and post-method hooks allow you to choose when you want to run logic in each phase. _Normally_ you will run logic in the "pre" hook but sometimes you may want to completely override the core logic or run logic that uses state that was updated _after_ the core logic runs.
+
+All update methods are passed an instance of the Excalibur engine, which
+can be used to perform coordinate math or access global state. It is also
+passed `delta` which is the time in milliseconds since the last frame, which can be used
+to perform time-based movement or time-based math (such as a [timer](/docs/utilities)).
+
+<docs-note>Reference [Actor lifecycle](#actor-lifecycle) for a breakdown of each phase and when things are executed.</docs-note>
+
+### Pre-update
+
+Override the [[Actor.onPreUpdate]] method to update the state of your actor before [[Actor.update]].
+**This is the typical method to override.** Things that need to be updated include state, drawing, or position.
+
+```ts
+class Player extends Actor {
+  /**
+   * Runs before "core" update logic, before this frame is updated
+   */
+  public onPreUpdate(engine: ex.Engine, delta: number) {
+    // update velocity
+    this.vel.setTo(-1, 0)
+  }
+}
+```
+
+### Core update
+
+[[Actor.update]] is the core update logic to prepare the frame. **This is an advanced method override.** You can take over the whole update loop by overriding this method. Use `super.update(engine, delta)` to invoke the core logic or leave it out to completely customize the update logic.
+
+```ts
+/**
+ * DANGER: This is for advanced users to totally override draw logic.
+ */
+public update(ctx: CanvasRenderingContext2D, delta: number) {
+
+  // Invoke the core logic
+  // Or leave it out to completely override
+  super.update(engine, delta);
+
+  // Perform custom update logic
+  // ...
+}
+```
+
+### Post-update
+
+[[Actor.onPostUpdate]] is called after [[Actor.update]] to prepare state for the _next_ frame.
+
+<docs-note>**Important:** At this time, the frame hasn't been drawn yet so state updated in this method will be reflected during the draw loop but will **not** be reflected in the scene graph until the _next_ frame.</docs-note>
```

Comment:
This is slightly confusing, this is still in the current frame just after the core update. The current frame draw will draw any state updated `onPostUpdate`

Reply:
I think what I am trying to say is that other actors in the scene will not see any changes you make during `onPostUpdate` in the same frame because they've already run their update loop. _Technically_ it probably is dependent on the order each `onPostDraw` is invoked within the loop.

Reply:
I think I've fixed it now, LMK what you think!